astropy/io/fits/hdu/hdulist.py

```diff
@@ -694,14 +694,13 @@ def writeto(self, fileobj, output_verify='exception', overwrite=False,
         # but only if the file doesn't exist.
         fileobj = _File(fileobj, mode='ostream', overwrite=overwrite)
         hdulist = self.fromfile(fileobj)
+        dirname = os.path.dirname(hdulist._file.name)
```

Comment:
`try-except` here does not hurt, if nothing else, at least it would be consistent with your other change above.